### PR TITLE
Feature/facemesh ios

### PR DESCRIPTION
**Pull request by Sufeenshaikh**
Facemesh extension for iOS

### CODE REVIEW

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

File: appinventor/components-ios/src/facemesh.swift [hunk 1]
```diff
@@ -0,0 +1,760 @@
...
+//        throw YailRuntimeError("Got unparsable array from Javascript", "RuntimeError")
+//      }
+//      return result
+//    }
+    
+    //      internal func assertWebView(_ method: String, _ frontFacing: Bool = true) throws {
+    //        guard let _webview = _webview else {
+    //          throw AIError.webviewerNotSet
+    //        }
+    //      }
```
Comment (daki7711) [inline appinventor/components-ios/src/facemesh.swift:131]: Let's erase all commented out code

File: appinventor/components-ios/src/facemesh.swift [hunk 2]
```diff
@@ -0,0 +1,760 @@
...
+    //          print(request)
+    //          _webview?.load(request)
+    //          print("Request loaded")
+    //        }else {
+    //          print("Error: Unable to find the HTML file in the bundle")
+    //        }
+    //        //      showImagePicker()
+    //      }
+    //    }
+    //  }
```
Comment (daki7711) [inline appinventor/components-ios/src/facemesh.swift:179]: Ditto

**Review by daki7711 — COMMENTED** [review]
Let's erase the comments first

File: appinventor/AICompanionApp.xcconfig.sample [hunk 3]
```diff
@@ -1,4 +0,0 @@
-// -*- mode: conf; -*-
-// Copyright 2023 MIT, All rights reserved.
-
-DEVELOPMENT_TEAM = "Your Team ID here"
```
Comment (daki7711) [inline appinventor/AICompanionApp.xcconfig.sample:4]: I think we need to restore this?

File: appinventor/AICompanionApp.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -352,6 +352,7 @@
 				TargetAttributes = {
 					06463EF71D95956800B23DCD = {
 						CreatedOnToolsVersion = 8.0;
+						DevelopmentTeam = A35W4MM59Y;
```
Comment (daki7711) [inline appinventor/AICompanionApp.xcodeproj/project.pbxproj:355]: I think all of these changes shouldn't have been added during github add, in a big project never do `git add .` . I know it's tedious but always do `git add <individual files you want to add>`

File: appinventor/components-ios/src/ImageSprite.swift [hunk 5]
```diff
@@ -58,7 +94,7 @@ open class ImageSprite: Sprite {
       }
     }
   }
-
+
```
Comment (daki7711) [inline appinventor/components-ios/src/ImageSprite.swift:97]: Clear out these minor changes and make the code consistent as possible

File: appinventor/components-ios/src/facemesh.swift [hunk 6]
```diff
@@ -0,0 +1,760 @@
...
+        let readAccessURL = Bundle(for: FaceExtension.self).bundleURL
+        let request = URLRequest(url: URL(string: "appinventor://localhost/\(assetPath ?? "").html")!)
+        //        let request = URLRequest(url: url)
+        print(request)
+        _webview?.load(request)
+        print("Request loaded")
+      }else {
+        print("request not loaded")
+      }
+            showImagePicker()
```
Comment (daki7711) [inline appinventor/components-ios/src/facemesh.swift:156]: can you align the number of shifts? in mac all you need to do is `command + a` and then `control + i`

File: appinventor/components-ios/src/Sprite.swift [hunk 7]
```diff
@@ -311,25 +379,25 @@ open class Sprite: ViewComponent, UIGestureRecognizerDelegate {
...
-        && normalizedAngle > 90 && normalizedAngle < 180)
-      || (e == Direction.southwest
-        && normalizedAngle > 180 && normalizedAngle < 270)
-      || (e == Direction.southeast && normalizedAngle > 270)) {
+                && normalizedAngle > 0 && normalizedAngle < 90)
+               || (e == Direction.northwest
+                   && normalizedAngle > 90 && normalizedAngle < 180)
+               || (e == Direction.southwest
+                   && normalizedAngle > 180 && normalizedAngle < 270)
+               || (e == Direction.southeast && normalizedAngle > 270)) {
```
Comment (daki7711) [inline appinventor/components-ios/src/Sprite.swift:400]: All these additional spaces seems unnecessary, can you restore them?

**Review by daki7711 — CHANGES_REQUESTED** [review]

File: appinventor/components-ios/src/facemesh.swift (hunk 1 above)
Comment (Sufeenshaikh) [reply appinventor/components-ios/src/facemesh.swift:131]: I've removed all the commented-out code

File: appinventor/AICompanionApp.xcodeproj/project.pbxproj (hunk 4 above)
Comment (Sufeenshaikh) [reply appinventor/AICompanionApp.xcodeproj/project.pbxproj:355]: Noted. I'll be more careful with git add in the future.

File: appinventor/components-ios/src/Sprite.swift (hunk 7 above)
Comment (Sufeenshaikh) [reply appinventor/components-ios/src/Sprite.swift:400]: I removed the extra spaces. They were probably there from before.

**Comment by Sufeenshaikh** [comment]
Sorry! I accidentally marked as ready for review

**Comment by daki7711** [comment]
> Sorry! I accidentally marked as ready for review

No need to sorry about this, I just did it to make sure we need review. Once things work we will convert it back to ready to review.
Please update the code as I suggested and push a new one

**Comment by Sufeenshaikh** [comment]
I've implemented the requested changes and pushed a new commit (913eb7f)

**Comment by ewpatton** [comment]
This is now on TestFlight.

**Comment by ewpatton** [comment]
@AppInventorWorkerBee ok to test

**Review by ewpatton — APPROVED** [review]

**Pull request merged**